core/variant/dictionary.cpp

```diff
@@ -39,10 +39,23 @@
 #include "core/variant/type_info.h"
 #include "core/variant/variant_internal.h"
 
+bool DictionaryVariantComparator::compare(const Variant &p_lhs, const Variant &p_rhs) {
+	if (p_lhs.hash_compare(p_rhs)) {
```

Comment:
this was taken from here:
https://github.com/godotengine/godot/blob/c2980b9c10b51f4a2aa6c3ded61d227cc18f029f/core/variant/variant.h#L796-L798
i wrote it like this to allow comparisons of different types between only String and StringName (because they are the only types that always hash the same way), if `OP_EQUAL` was used here, it would change the behavior in more ways (as in an empty `PackedFloat32Array` and an empty `PackedFloat64Array` might be equivalent keys), and that doesn't seem correct to me or it at least doesn't seem like it belongs in this PR

Reply:
`hash_compare` is the same as the Variant `==` operator, it doesn't compare hashes and so it doesn't have collisions
it's equivalent to what it replaced, while also allowing comparisons between String <-> StringName
i don't think this is incorrect

Reply:
You are right, I guess the function name is not the best.